pimd/pim_upstream.c

```diff
@@ -2062,6 +2062,7 @@ static bool pim_upstream_sg_running_proc(struct pim_upstream *up)
 	if ((up->sptbit != PIM_UPSTREAM_SPTBIT_TRUE) &&
 	    (up->rpf.source_nexthop.interface)) {
 		pim_upstream_set_sptbit(up, up->rpf.source_nexthop.interface);
+		pim_upstream_update_could_assert(up);
```

Comment:
Could we get more data here in the commit message about what is going on and why this is good to change?

Reply:
I have added additional information in the commit messages

Reply:
I have merged these two patches related to PIM assert and verified them in the following ring topology. However, there are still issues with PIM assert as downstream routers continue to receive duplicate multicast packets. I have submitted a new bug report（bugid: 13574） with specific details on the issue. Could you please take some time to help me review it?
```
Build the following network topology
										Receiver
										   |
										   |
									 br-lan_R_3403				
								                   |
										   |
										Router2
									  /			\
									/			  \
							br-lan_R_3401  		     br-lan_R_3402
								    /				      \
								  /					\
								/					  \
						br-lan_R_3401				              \
							    /					              \
							  /                    		                        \				
Sender ---br-lan_R_3403--	Router1             		                      Switch------ br-lan_R_3401--Router4--br-lan_R_3402--- Receiver1				
            				                \						       /
							  \						     /
							br-lan_R_3402			           /	
	      	  					      \                  	                 /
								\				       /
								  \				     /
								br-lan_R_3401	br-lan_R_3402	
									\			 /
									  \		       /
									    Router3 	

sender: 192.168.10.2
receiver1: 192.168.40.2
receiver2: 192.168.20.2

Router1: 
br-lan_R_3401: 192.168.122.1
br-lan_R_3402: 192.168.144.1
br-lan_R_3403: 192.168.10.1

Router2：
br-lan_R_3401: 192.168.122.2
br-lan_R_3402: 192.168.155.1
br-lan_R_3403: 192.168.20.1

Router3:
br-lan_R_3401: 192.168.144.2
br-lan_R_3402: 192.168.155.2

Router4：
br-lan_R_3401: 192.168.155.3
br-lan_R_3402: 192.168.155.2
```